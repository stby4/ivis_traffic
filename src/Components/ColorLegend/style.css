.mapLegend {
    display: inline-block;
    line-height: 16px;
    margin-bottom: 20px;
}

.mapLegendMin {
    margin-right: 5px;
}

.mapLegendMax {
    margin-left: 5px;
}