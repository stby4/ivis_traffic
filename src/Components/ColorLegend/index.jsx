import React, { PureComponent } from 'react'

import './style.css'

class ColorLegend extends PureComponent {

    render() {
        const { color, domain, width, height } = this.props

        const cols = Array(width).fill(0).map((x, y) => x + y)
        const colorStep = 1 / width


        const min = domain[0]
        const max = domain[1]

        const invert = domain[0] > domain[1]

        return (
            <div className="mapLegend">
                <span className="mapLegendMin">{Number(invert ? max : min).toLocaleString()}</span>
                <svg width={width} height={height} >
                    {
                        cols.map(col => <path key={`path${col}`} d={`M${col} 0 V ${height}`} stroke={`${color(invert ? 1 - col * colorStep : col * colorStep)}`} strokeWidth="2" />)
                    }
                </svg>
                <span className="mapLegendMax">{Number(invert ? min : max).toLocaleString()}</span>
            </div>
        )
    }
}

export default ColorLegend