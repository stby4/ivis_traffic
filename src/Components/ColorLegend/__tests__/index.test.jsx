import React from 'react'
import renderer from 'react-test-renderer'
import ColorLegend from '../'

const color = jest.fn()
const domain = [0, 2000]

test('Snapshot ColorLegend', () => {
    const tree = renderer
        .create(
            <ColorLegend color={color} domain={domain} width="500" height="100" />
        )
        .toJSON()
    expect(tree).toMatchSnapshot()
    expect(color.mock.calls.length).toBe(1)
})