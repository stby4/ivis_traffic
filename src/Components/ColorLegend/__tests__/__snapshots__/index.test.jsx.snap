// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot ColorLegend 1`] = `
<div
  className="mapLegend"
>
  <span
    className="mapLegendMin"
  >
    0
  </span>
  <svg
    height="100"
    width="500"
  >
    <path
      d="M0 0 V 100"
      stroke="undefined"
      strokeWidth="2"
    />
  </svg>
  <span
    className="mapLegendMax"
  >
    2,000
  </span>
</div>
`;
